hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -343,6 +362,45 @@ func MachineDeploymentComplete(deployment *capiv1.MachineDeployment) bool {
 		newStatus.ObservedGeneration >= deployment.Generation
 }
 
+func generateMachineHealthCheck(nodePool *hyperv1.NodePool, targetNamespace, infraID string) *capiv1.MachineHealthCheck {
+	maxUnhealthy := intstr.FromInt(2)
+	resourcesName := generateName(infraID, nodePool.Spec.ClusterName, nodePool.GetName())
+	return &capiv1.MachineHealthCheck{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nodePool.GetName(),
+			Namespace: targetNamespace,
+		},
+		// Opinionated spec based on https://github.com/openshift/managed-cluster-config/blob/14d4255ec75dc263ffd3d897dfccc725cb2b7072/deploy/osd-machine-api/011-machine-api.srep-worker-healthcheck.MachineHealthCheck.yaml
+		// TODO (alberto): possibly expose this config at the nodePool API.
+		Spec: capiv1.MachineHealthCheckSpec{
+			ClusterName: nodePool.Spec.ClusterName,
+			Selector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					resourcesName: resourcesName,
+				},
+			},
+			UnhealthyConditions: []capiv1.UnhealthyCondition{
+				{
+					Type:   "Ready",
+					Status: "false",
```

Comment:
Is there a constant for this, and is the value case sensitive? Not sure I've ever seen a lowercase status like this elsewhere.